### PR TITLE
Logback in all tests

### PR DESCRIPTION
**Pull request by mbaechler**

### CODE REVIEW

**Comment by mbaechler** [comment]
It replaces https://github.com/linagora/james-project/pull/956 2 years later

File: testing/base/src/main/resources/logback-test.xml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<?xml version="1.0" encoding="UTF-8" debug="true" ?>
```
Comment (rouazana) [inline testing/base/src/main/resources/logback-test.xml]: unrelated?

Comment (mbaechler) [reply testing/base/src/main/resources/logback-test.xml]: well, it's one commit too early but it doesn't hurt

**Comment by chibenwa** [comment]
Shouldn't you remove junit 4.12 dependencies every where as testing base brings vintage in?

**Comment by mbaechler** [comment]
> Shouldn't you remove junit 4.12 dependencies every where as testing base brings vintage in?

Last time I tried that, it didn't work because junit brings more than what is in vintage.

I could remove vintage from base and import it only where necessary but that would partly defeat the simplicity of current solution.

Tell me what you think

**Comment by mbaechler** [comment]
> Problem creating jar: No space left on device

So bad.

**Comment by mbaechler** [comment]
test this please

**Comment by chibenwa** [comment]
Here is what I succeeded to do. (Fully remove junit:junit::4.12

As you were cautious, I did run tests in every single modified module (maven) to ensure things go fine.

Ah, and BTW this even makes things faster (from 5min30 to 4min57 on my local computer).

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
test this please

ci 1 is not meaningful for measuring build time improvment.

**Comment by chibenwa** [comment]
```
[1fd6bee680d144be81997233fa5e9ef4c94ac2c2] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-jar-plugin:3.1.2:jar (default-jar) on project blob-objectstorage: Error assembling JAR: Problem creating jar: No space left on device -> [Help 1]
```

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
```
[1fd6bee680d144be81997233fa5e9ef4c94ac2c2] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-jar-plugin:3.1.2:jar (default-jar) on project blob-objectstorage: Error assembling JAR: Problem creating jar: No space left on device -> [Help 1]
```

test this please

**Comment by chibenwa** [comment]
test this please

**Comment by chibenwa** [comment]
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-jar-plugin:3.1.2:jar (default-jar) on project blob-objectstorage: Error assembling JAR: Problem creating jar: No space left on device -> [Help 1]
```

Yet again

test this please

**Comment by chibenwa** [comment]
Total build output: 140,276 KB

**Comment by chibenwa** [comment]
`CassandraMailRepositoryTest.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult » Execution`

test this please

**Comment by chibenwa** [comment]
Merged
